### ISSUE TITLE
ShellContextMenu in 3.3.1 throw AccessViolationException

### ISSUE THREAD

**Issue by zhuxb711** [dahall/Vanara#195]
**Describe the bug**

![image](https://user-images.githubusercontent.com/42038175/105273400-311c4f80-5bd6-11eb-9657-fe8bd596ce02.png)

**What code is involved**
```csharp
using (ShellItem Item = ShellItem.Open(Path))
using (ShellContextMenu Menu = new ShellContextMenu(Item))
{
      foreach (ShellContextMenu.MenuItemInfo MenuItem in Menu.GetItems())
      {
             //Do something
      }
}
```

**Expected behavior**
Do not throw an exception

**Comment by zhuxb711** [comment]
Call stack:
![image](https://user-images.githubusercontent.com/42038175/105274247-c53ae680-5bd7-11eb-808e-c9d34945407c.png)

**Comment by zhuxb711** [comment]
at Vanara.PInvoke.Shell32.IContextMenu.GetCommandString(IntPtr idCmd, GCS uType, IntPtr pReserved, IntPtr pszName, UInt32 cchMax)
   at Vanara.Windows.Shell.ShellContextMenu.GetCommandString(Int32 command, GCS stringType)
   at Vanara.Windows.Shell.ShellContextMenu.MenuItemInfo.GetMenuItems(HMENU hMenu, ShellContextMenu scm)
   at Vanara.Windows.Shell.ShellContextMenu.GetItems(CMF menuOptions)

**Comment by dahall** [comment]
Sorry. I'll get it fixed and push an update tomorrow.

**Comment by zhuxb711** [comment]
Just for curious. What is the root cause of this issue?

**Comment by dahall** [comment]
I found a minor bug in how I was counting characters, but before I fixed that, I could not reproduce the error you're seeing. What is the extension of the file you are passing to `ShellItem`?

**Comment by dahall** [comment]
Nevermind, I found a way to reproduce. Troubleshooting now.

**Comment by dahall** [comment]
It is now fixed and I'll push 3.3.2 shortly. This isn't a new problem so I'm surprised testing hasn't found it before. Effectively, GetCommandString blows up if the menu item is not a string (e.g. separators). I was checking for the flag indicating string, but the value of that flag is 0, so it was always true. I need to check for equality. This also helped me find a non-breaking problem with my allocation of memory, so I got a two-for.
